src/test/java/wolox/training/controllers/BookControllerTests.java

```diff
@@ -0,0 +1,246 @@
+package wolox.training.controllers;
+
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.Mockito.doNothing;
+import static org.mockito.Mockito.doThrow;
+import static org.springframework.test.web.servlet.request.MockMvcRequestBuilders.delete;
+import static org.springframework.test.web.servlet.request.MockMvcRequestBuilders.get;
+import static org.springframework.test.web.servlet.request.MockMvcRequestBuilders.post;
+import static org.springframework.test.web.servlet.request.MockMvcRequestBuilders.put;
+import static org.springframework.test.web.servlet.result.MockMvcResultMatchers.content;
+import static org.springframework.test.web.servlet.result.MockMvcResultMatchers.status;
+
+import java.time.LocalDateTime;
+import java.util.Optional;
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.Test;
+import org.mockito.Mockito;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.test.autoconfigure.web.servlet.WebMvcTest;
+import org.springframework.boot.test.mock.mockito.MockBean;
+import org.springframework.http.MediaType;
+import org.springframework.test.web.servlet.MockMvc;
+import wolox.training.exceptions.IdNotFoundException;
+import wolox.training.exceptions.NullFieldException;
+import wolox.training.models.Book;
+import wolox.training.repositories.BookRepository;
+import wolox.training.validators.BookValidator;
+
+@WebMvcTest(controllers = BookController.class)
+public class BookControllerTests {
+
+    private static Book book1;
+    private static String bookWithVariableYearJsonString;
+    private static String bookRandomJsonString;
+    private static String basicUrl;
+    @Autowired
+    private MockMvc mvc;
+    @MockBean
+    private BookRepository bookRepository;
+    @MockBean
+    private BookValidator bookValidator;
+
+    @BeforeAll
+    static void setUp() {
+        basicUrl = "/api/books/";
+
+        bookWithVariableYearJsonString = "{ "
```

Comment:
Hello! It's ok, but it's easier to use the ObjectMapper.writeValueAsString(object) to build a JsonString. 😁

Reply:
Done :D